src/mad_spark_alt/core/smart_orchestrator.py

```
@@ -484,27 +484,68 @@
+                # Cancel pending tasks to clean up resources
+                for task in pending:
+                    task.cancel()
+
+                # Wait briefly for cancellation to complete
+                await asyncio.gather(*pending, return_exceptions=True)
+
+                # Merge completed results with timeout placeholders
+                result_dict = {}
+                task_list = list(done) + list(pending)
+
+                for i, (method, task) in enumerate(zip(available_methods, tasks)):
```

Ruff (0.11.9)

Loop control variable `i` not used within loop body

Rename unused `i` to `_i`

(B007)